src/seedu/addressbook/Main.java

```diff
@@ -81,14 +81,20 @@ private void exit() {
     /** Reads the user command and executes it, until the user issues the exit command.  */
     private void runCommandLoopUntilExitCommand() {
         Command command;
-        do {
-            String userCommandText = ui.getUserCommand();
-            command = new Parser().parseCommand(userCommandText);
-            CommandResult result = executeCommand(command);
-            recordResult(result);
-            ui.showResultToUser(result);
-
-        } while (!ExitCommand.isExit(command));
+        try {
```

Comment:
Try block should contain at little code as possible, i.e. catch the exception at the exact line of possible occurrence. Irrelevant lines should not be inside the try block.